addons/payment/models/payment_token.py

```diff
@@ -83,8 +83,14 @@ def write(self, values):
         """
         if 'active' in values:
             if values['active']:
-                if any(not token.active for token in self):
-                    raise UserError(_("A token cannot be unarchived once it has been archived."))
+                if any(
+                    not token.payment_method_id.active
+                    or token.provider_id.state == "disabled"
+                    for token in self
+                ):
+                    raise UserError(
+                        _("Once you archive the payment method and provider, you can't unarchive the token.")
```

Comment:
I think message is wrong :(
it should be
`You can't unarchive token linked to inactive payment method or disable provider`
@dit-odoo could you validate error message? :smiley:

Reply:
@kcv-odoo

Hello,
For provider:  " Disabling the provider will also archive 2 tokens associated with it " 
For tokens: ok with what you suggest, "You can't unarchive tokens linked to inactive payment methods or disabled providers"

Reply:
Done the changes :+1: 

Thankyou so much :smile: